src/acl/acl.test.ts

```diff
@@ -310,6 +354,66 @@ describe("fetchResourceAcl", () => {
     expect(mockFetch.mock.calls).toHaveLength(1);
     expect(mockFetch.mock.calls[0][0]).toBe("https://some.pod/resource.acl");
   });
+
+  it("throws an error if the linked Resource is an ACP ACR", async () => {
+    const sourceDataset: WithServerResourceInfo = {
+      internal_resourceInfo: {
+        sourceIri: "https://some.pod/resource",
+        isRawData: false,
+        aclUrl: "https://some.pod/resource.acl",
+        linkedResources: {
+          acl: ["https://some.pod/resource.acl"],
+        },
+      },
+    };
+    const mockFetch = jest.fn(window.fetch).mockResolvedValueOnce(
+      mockResponse(undefined, {
+        url: "https://some.pod/resource?ext=acr",
+        headers: {
+          "Content-Type": "text/turtle",
+          Link: '<http://www.w3.org/ns/solid/acp#AccessControlResource>; rel="type"',
+        },
+      })
+    );
+
+    await expect(
+      internal_fetchResourceAcl(sourceDataset, {
+        fetch: mockFetch,
+      })
+    ).rejects.toThrow(
+      "[https://some.pod/resource] is governed by Access Control Policies in [https://some.pod/resource?ext=acr] rather than by Web Access Control."
+    );
+  });
+
+  it("throws an error if the linked Resource is an ACP ACR and thus has another type", async () => {
```

Comment:
I'm not sure about the phrasing of this test description. If I get it right, the intent is to test that `internal_fetchResourceAcl` throws if dereferencing the target resource return several Link headers, one of them indicating that it is an ACR. Is that correct ? If so, what happens if one of the Link headers describes the resource as governed by an ACL as well (in other words, if the Link headers are contradictory) ?

I mean, the current implementation throws, and that may be the behaviour that we want, but it's not entirely explicit.

Reply:
Hmm, that's a good question... The problem is that there is no Link header that describes a Resource as an ACL (which is why we only check if it's _not_ an ACR). But yes, if the Link headers are contradictory, I think we'd still want to throw, since if something so idiosyncratic is the case, we cannot reliably assume that it's an ACL. I've updated the test description to no longer conclude that it's definitely not an ACL if it's advertised as an ACR: dc3f9ffc